src/pset/mod.rs

```diff
@@ -180,19 +178,19 @@ impl PartiallySignedTransaction {
 
     /// Extract the Transaction from a PartiallySignedTransaction by filling in
     /// the available signature information in place.
-    pub fn extract_tx(self) -> Result<Transaction, Error> {
+    pub fn extract_tx(&self) -> Result<Transaction, Error> {
         // This should never trigger any error, should be panic here?
         self.sanity_check()?;
         let locktime = self.locktime()?;
         let mut inputs = vec![];
         let mut outputs = vec![];
 
-        for psetin in self.inputs.into_iter() {
+        for psetin in self.inputs.iter() {
             let txin = TxIn {
                 previous_output: OutPoint::new(psetin.previous_txid, psetin.previous_output_index),
                 is_pegin: psetin.previous_output_index & (1 << 30) != 0,
                 has_issuance: psetin.previous_output_index & (1 << 31) != 0,
-                script_sig: psetin.final_script_sig.unwrap_or(Script::new()),
+                script_sig: psetin.final_script_sig.clone().unwrap_or(Script::new()),
```

Comment:
I know this is the same as the old code, but FYI we could use `unwrap_or_default` here (and a few other places)